data_validation/clients.py

```diff
@@ -274,6 +274,23 @@ def get_max_column_length(client):
     return 128
 
 
+def get_max_in_list_size(client, in_list_over_expressions=False):
+    if client.name == "snowflake":
+        if in_list_over_expressions:
+            # This is a workaround for Snowflake limitation:
+            #   SQL compilation error: In-list contains more than 50 non-constant values
+            # getattr(..., "cast") expression above is looking for lists where the contents are casts and not simple literals.
+            return 50
+        else:
+            return 16000
```

Comment:
Curious to know: how did you find this other number for the list size? It would be helpful to put some sort of product documentation as a comment for future reference, but I could only find community posts mentioning it like this one: https://community.snowflake.com/s/question/0D5Do00000O3sJ0KAJ/how-o-get-around-th-16k-limit-on-number-of-expressions-in-in-clause-in-sql-generated-by-a-businessobjects-report

Reply:
It was not the same post but I saw another 2 Snowflake community posts.

I don't think we need to document it because this code is ensuring that we don't hit the 16k limit, at 16,000 we'll switch to:

```
WHERE id IN (1, ..., 16000) OR id IN (16001, ..., n)
```

We will find other limits for other engines I'm sure. I started to read about SQL Server but that has a memory limit based on the execution plan and not a hard/fixed limit, so I decided to just include the limits I knew about for now.

Reply:
Right, it makes sense, it's clearer for me now, thank you! 

And for sure, totally agree that we might face it for other engines but at the moment we can stick with the ones we're aware of the values and usage